# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: toolkit/components/cookiebanners/nsCookieInjector.cpp
# Commit: bb99a3d0d1e7
# Full Hash: bb99a3d0d1e7bf2278834e61e8878440bca791e3
# Author: Valentin Gosu <valentin.gosu@gmail.com>
# Date: 2025-03-29 09:19:43
# Regressor Bug: 1951750
# File Overlap Count: 4
# Description:
#   Bug 1951750 - Handle domain argument in CookieStoreParent r=baku,edgul,cookie-reviewers
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D240695
# ==============================================================================

diff -r ee233c9ac108 -r bb99a3d0d1e7 toolkit/components/cookiebanners/nsCookieInjector.cpp
--- a/toolkit/components/cookiebanners/nsCookieInjector.cpp	Fri Mar 28 13:48:31 2025 +0000
+++ b/toolkit/components/cookiebanners/nsCookieInjector.cpp	Fri Mar 28 13:48:31 2025 +0000
@@ -324,10 +324,11 @@
             ("Setting cookie: %s, %s, %s, %s\n", c.Host().get(), c.Name().get(),
              c.Path().get(), c.Value().get()));
     rv = cookieManager->AddNative(
-        c.Host(), c.Path(), c.Name(), c.Value(), c.IsSecure(), c.IsHttpOnly(),
-        c.IsSession(), c.Expiry(), &aOriginAttributes, c.SameSite(),
-        static_cast<nsICookie::schemeType>(c.SchemeMap()),
-        /* is partitioned: */ false, nullptr);
+        nullptr, c.Host(), c.Path(), c.Name(), c.Value(), c.IsSecure(),
+        c.IsHttpOnly(), c.IsSession(), c.Expiry(), &aOriginAttributes,
+        c.SameSite(), static_cast<nsICookie::schemeType>(c.SchemeMap()),
+        /* is partitioned: */ false, nullptr,
+        [](mozilla::net::CookieStruct&) { return true; });
     NS_ENSURE_SUCCESS(rv, rv);
 
     aHasInjectedCookie = true;
